### DISCUSSION TITLE
How can I stop this error. LF will be replaced by CRLF the next time Git touches it

### DISCUSSION THREAD

**Discussion by Blockheded** [community/community#66838] [New to GitHub] [answered]
The person I am working with does not have this issue when committing to main, but I do, and I have no clue how to fix it.

LF will be replaced by CRLF the next time Git touches it
 error: open("Temp/FSTimeGet-7ef7d120c5d912643bdf9366b1285a3d"): Permission denied 
fatal: Unable to process path Temp/FSTimeGet-7ef7d120c5d912643bdf9366b1285a3d

**Comment by seyLu** [comment]
#### Hi @Blockheded,

> LF will be replaced by CRLF the next time Git touches it

This is normal. This has something to do with line endings, which are different in Linux/Windows/Mac. Linux uses LF, Windows uses CRLF -- completely normal thing.

> error: open("Temp/FSTimeGet-7ef7d120c5d912643bdf9366b1285a3d"): Permission denied
fatal: Unable to process path Temp/FSTimeGet-7ef7d120c5d912643bdf9366b1285a3d

The error message here is not related to the statement before it. Temp directories shouldn't be tracked by git, and the way to do it is to add `[T]emp` in .gitignore file.

Most prolly you are using Unity with Git, so you can use the [.gitignore template in Unity](https://github.com/github/gitignore/blob/main/Unity.gitignore) instead. 

Also make sure the project structure is:

```bash
Assets
Library
Temp
...
.gitignore
```

and not:

```bash
UnityProject
  |-- Assets
  |-- Library
  |-- Temp
  |-- ...
.gitignore 
```

or else the .gitignore will not work.

**Reply by faulknor82** [reply]
Thanks. You're a life saver. I've been unable to push anything to GitHub for months. My problem was your last suggestion. My .gitignore file was one folder up. I moved it into my project folder to be on the same level as my Assets folder and now everything works.

**Comment by andreslopezferro** [comment, marked as answer]
This error message is Git's way of warning you that it's going to automatically change line endings in your files to match the expected format for your operating system. It's important to have consistent line endings in a project to avoid conflicts and unexpected behavior.

Configure Git to handle line endings correctly for your operating system by setting the `core.autocrlf` configuration option.
**For Windows**:
`git config --global core.autocrlf true`
**For macOS/Linux**:
`git config --global core.autocrlf input`
The "_true_" setting instructs Git to convert line endings to CRLF when checking out files and to LF when committing files.
In contrast, the "_input_" setting directs Git to convert line endings to LF during commits while preserving them as is during checkouts.


Hope this can help you!

**Reply by dbc60** [reply]
I really disagree with allowing Git to change line endings automatically. If you must have a `.gitattributes` file, then please use `* text=auto eol=lf` so every text file defaults to LF. This way, only Windows-specific files (like `*.bat`, `*.cmd`, `*.ps1`, `sln`, and `*.vcxproj` and maybe a few others) need `eol=crlf` applied.

**Reply by isimmons** [reply]
Any other settings for this? I've been ignoring it for a while because it didn't work for me but it is irritating to see the warning.
`git config --global core.autocrlf` outputs true
And still the warning shows.

**Reply by isimmons** [reply]
A lot of settings affect this. I set autocrlf to false and used the .gitatttributes method dbc60 suggests. Then I started seeing the same warning message but in reverse 'CRLF will be converted to LF'. I have a .editorconfig with end_of_line for all files set to LF. Then I found that PHPStorm is ignoring this. So I had to set the eol in PHPStorm settings for LF. So I'll probably have to set CRLF per file for Windows specific files.

**Reply by wildansuksesamin** [reply]
Thank you so muchh brotherrr

**Reply by usereri** [reply]
The command wasn't working for me. I tried it and checked if anything changed in config file, but there was no change. I just put `autocrlf = true` under `[core]` section and it worked for me.

**Reply by constup** [reply]
> The "true" setting instructs Git to convert line endings to CRLF when checking out files and to LF when committing files.

Have a Bash script in your project, or have a Bash project and there will be issues. Bash scripts require Unix style line endings to work properly, even on Windows. So, when you are working on some scripts, on Windows and on `develop` branch and keep committing, everything will work just fine. As soon as you checkout to `master`, merge and push the changes, and return to `develop`, every single Bash script will get broken with CRLF.

**Comment by Ange290** [comment]
This error message is Git's way of warning you that it's going to automatically change line endings in your files to match the expected format for your operating system. It's important to have consistent line endings in a project to avoid conflicts and unexpected behavior.

Configure Git to handle line endings correctly for your operating system by setting the core.autocrlf configuration option.
For Windows:
git config --global core.autocrlf true
For macOS/Linux:
git config --global core.autocrlf input
The "true" setting instructs Git to convert line endings to CRLF when checking out files and to LF when committing files.
In contrast, the "input" setting directs Git to convert line endings to LF during commits while preserving them as is during checkouts.

Hope this can help you!

 that is true was help me

**Reply by sulth1** [reply]
Hi, total noob here. How do I do that? I've tried adding "git config --global core.autocrlf true" in the ".gitconfig" file, but it results in fatal: bad config line.

**Reply by speedy-sasquatch** [reply]
`git config --global core.autocrlf true` is not a config line, it’s the command you run to add the config line. Try running that command on your command line. When you do, it should add a line like `autocrlf = true` to the `[core]` section of your `.gitconfig` file.

Also, make sure you’re looking for the right config file because there are often at least three of them: a system-wide one, a user-wide (aka global) one, and a per-repository (aka local) one. Since you’re using the `--global` option in your command, the one you’re looking for should be located in your user home directory.

**Comment by drakulis** [comment]
`core.autocrlf` set to `true` by default is the most ridiculous decision. It should be set to `false` by default.
I am a developer and I know what I'm doing. I don't need Git messing around with my files without my knowledge. - it brings just headache and more undesirable effects than good ones.

There... I just had to say it. Thank you.

**Comment by MYeager1967** [comment]
```> git add -A -- .
"C:\\Program Files\\Git\\cmd\\git-crypt.exe" clean: line 1: C:\Program Files\Git\cmd\git-crypt.exe: No such file or directory
error: external filter '"C:\\Program Files\\Git\\cmd\\git-crypt.exe" clean' failed 127
error: external filter '"C:\\Program Files\\Git\\cmd\\git-crypt.exe" clean' failed```

Why is this command getting the extra \ everywhere? It's erroring out on the CR/CRLF thing...
